bson/src/main/org/bson/FieldNameValidator.java

```diff
@@ -30,6 +32,20 @@ public interface FieldNameValidator {
      */
     boolean validate(String fieldName);
 
+    /**
+     * Return the validation error message for an invalid field
+     *
+     * @param fieldName the field name
+     * @return the validation error message
+     * @throws IllegalArgumentException if fieldName is actually valid
+     */
+    default String getValidationErrorMessage(final String fieldName) {
+        if (validate(fieldName)) {
+            throw new IllegalArgumentException(format("%s is valid", fieldName));
```

Comment:
Is this tested?

Reply:
Ah, I see what you mean.  No, I didn't test it because `AbstractBsonWriter` guarantees that this will never happen.  Because of that it is more appropriate to use the Assertions class here instead to reflect the intent (and we typically don't test assertions).